### ISSUE TITLE
[Bug]: email activation for new users can't be turned of for private registration via API

### ISSUE THREAD

**Issue by petzsch** [btcpayserver/btcpayserver#6406]
### What is your BTCPay version?

BTCPay Server v2.0.3+558f45a48

### How did you deploy BTCPay Server?

docker

### What happened?

A [user on Telegram](https://t.me/btcpayserver/81254) reported that he's missing an API call of activating newly created user accounts (with API created).
After some research by @pavlenex and myself we found that the Option "Email confirmation required" can only be disabled if "Enable public user registration" is on in server settings. However there should be no self registration enabled as users are created via the API.

It seems that this use case was overlooked here.

I think it would be better to allow toggle of email confirmation requirement independently if public registration is enabled.

### How did you encounter this bug?

I didn't, the above mentioned telegram user did.

### Relevant log output

```shell
none
```

### What browser do you use?

not specified

### Additional information

none

### Are you sure this is a bug report?

- [X] I confirm this is a bug report

**Comment by pavlenex** [comment]
> I think it would be better to allow toggle of email confirmation requirement independently if public registration is enabled.

I don't think this should be a desired outcome, and it wouldn't fix the underlaying problem. There should be some sort of explanation in the API or the error that this needs to be enabled, since from what I understand ,the user leveraging API had no context on what's happening.

**Comment by lightningcheckout** [comment]
I wasn't aware of this issue created. 
After your explanation on TG that the activation of the user only impacts GUI, this is not an issue for me. 
It took some time to figure this out, so perhaps some additional info in the docs could be of help. 
Or an option to enable the account (via an additional field or automatically) when password is provided.

**Comment by dennisreimann** [comment]
@lightningcheckout Yes, these are expected to behave this way. I've added #6425 to provide more precise error messages for the approval endpoint to distinguish why approval failed.

**Comment by Mecanik** [comment]
This issue is currently happening to myself with the latest version. I wish to:

- Have public registrations disabled
- Create users via API

However, what happens is exactly what the OP described. I create a user via the API and email confirmation gets sent out. The only current workaround I can see is to break/remove the SMTP configuration details... which I really don't want to do, because I wish to receive email notifications myself.

What is the solution for this? Email confirmations cannot be turned off unless public registration is ON.

Thanks

**Comment by Mecanik** [comment]
Help?

**Comment by rockstardev** [comment]
@Mecanik help is on the way 😂

**Comment by Mecanik** [comment]
@rockstardev Sorry for being a nab here, how can I get the changes? I installed via btcpay-setup.sh.

**Comment by rockstardev** [comment]
@Mecanik I need to sync with @pavlenex and @NicolasDorier on when we're releasing the new version. In the meantime, I have tagged `2.1.2-rc3` for you, so you can deploy this fix by doing:

export BTCPAY_IMAGE="btcpayserver/btcpayserver:2.1.2-rc3"

And then just run `. btcpay-setup.sh -i` again, and you'll be set. You'll need to monitor when we release a new version so you clear BTCPAY_IMAGE var, can just do it when it's time with `unset BTCPAY_IMAGE` or `export BTCPAY_IMAGE=""`.

**Comment by Mecanik** [comment]
> [@Mecanik](https://github.com/Mecanik) I need to sync with [@pavlenex](https://github.com/pavlenex) and [@NicolasDorier](https://github.com/NicolasDorier) on when we're releasing the new version. In the meantime, I have tagged `2.1.2-rc3` for you, so you can deploy this fix by doing:
> 
> export BTCPAY_IMAGE="btcpayserver/btcpayserver:2.1.2-rc3"
> 
> And then just run `. btcpay-setup.sh -i` again, and you'll be set. You'll need to monitor when we release a new version so you clear BTCPAY_IMAGE var, can just do it when it's time with `unset BTCPAY_IMAGE` or `export BTCPAY_IMAGE=""`.

Thank you very much. I hope release is coming soon so I am officially up to date.

**Comment by Mecanik** [comment]
I did what you instructed but it's still v2.1.1+fd5800a82

**Comment by mp21-org** [comment]
My not-updated version number is v2.1.1+6b5cab65a  (different hash).
So perhaps try if you issue has been solved in your version - it seems the version number hasn't been increased for this RC release.

**Comment by Mecanik** [comment]
Yes, resolved. Thanks.

**Comment by Mecanik** [comment]
Thank you, all sorted!
